plan/logical_plan_builder.go

```diff
@@ -1212,43 +1213,94 @@ func (b *planBuilder) buildDataSource(tn *ast.TableName) LogicalPlan {
 			pkCol = schema.Columns[schema.Len()-1]
 		}
 	}
+	p.SetSchema(schema)
```

Comment:
I think `DataSource` don't need to know which columns are generated or not. And in `column prune` phase, those useless columns (include generated columns) will be pruned.